File: Controller/InfoCtrl.js
import Info from "../Model/Info.js";

export default class InfoCtrl {
    // Traduzir comandos http em ações negociais
    // Conceito REST
    // Considerar o protocolo HTTP

    gravar(requisicao, resposta) {
        if(requisicao.method == "POST" && requisicao.is("application/json")) {
            const dados = requisicao.body;

            // pseudo validação
            if (
                dados.nome && dados.descricao && dados.status && dados.urlImagem
            ) {
                const servico = new Info(
                    0,
                    dados.nome,
                    dados.descricao,
                    dados.status,
                    dados.urlImagem
                );

                servico.gravar()
                    .then(() => {
                        resposta.status(201).json({
                            "status": true,
                            "mensagem": "Info gravada com sucesso!",
                            "id": servico.id
                        });
                    })
                    .catch((error) => {
                        resposta.status(500).json({
                            "status": false,
                            "mensagem": "Erro ao registrar a info: " + error.message
                        });
                    });
            } else {
                resposta.status(400).json({
                    "status": false,
                    "mensagem": "Informe todos os dados necessários conforme documentação!"
                });
            }
        } else {
            resposta.status(405).json({
                "status": false,
                "mensagem": "Formato não permitido!"
            });
        }
    }

    alterar(requisicao, resposta) {
        if(["PUT","PATCH"].includes(requisicao.method) && requisicao.is("application/json")) {
            const dados = requisicao.body;

            // pseudo validação
            if (
                dados.id && dados.nome && dados.descricao && dados.status  && dados.urlImagem
            ) {
                const servico = new Info(
                    dados.id,
                    dados.nome,
                    dados.descricao,
                    dados.status,
                    dados.urlImagem
                );

                servico.alterar()
                    .then(() => {
                        resposta.status(200).json({
                            "status": true,
                            "mensagem": "Info alterada com sucesso!"
                        });
                    })
                    .catch((error) => {
                        resposta.status(500).json({
                            "status": false,
                            "mensagem": "Erro ao alterar a info: " + error.message
                        });
                    });
            } else {
                resposta.status(400).json({
                    "status": false,
                    "mensagem": "Informe todos os dados necessários conforme documentação!"
                });
            }
        } else {
            resposta.status(405).json({
                "status": false,
                "mensagem": "Formato não permitido!"
            });
        }
    }
    
    excluir(requisicao, resposta) {
        if(requisicao.method == "DELETE" && requisicao.is("application/json")) {
            const id = requisicao.params.id; // O id deve ser informado na url

            // pseudo validação
            if (id > 0) {
                const info = new Info(id);

                info.excluir()
                    .then(() => {
                        resposta.status(200).json({
                            "status": true,
                            "mensagem": "Info excluída com sucesso!"
                        });
                    })
                    .catch((error) => {
                        resposta.status(500).json({
                            "status": false,
                            "mensagem": "Erro ao excluir a info: " + error.message
                        });
                    });
            } else {
                resposta.status(400).json({
                    "status": false,
                    "mensagem": "Informe o id na url!"
                });
            }
        } else {
            resposta.status(405).json({
                "status": false,
                "mensagem": "Formato não permitido!"
            });
        }
    }
    
    consultar(requisicao, resposta) {
        const termoBusca = requisicao.params.info;
        if(requisicao.method == "GET") {
            const info = new Info(0);

            info.consultar(termoBusca)
                .then((listaInfos) => {
                    resposta.status(200).json({
                        "status": true,
                        listaInfos
                    });
                })
                .catch((error) => {
                    resposta.status(500).json({
                        "status": false,
                        "mensagem": "Não foi possível recuperar as infos: " + error.message
                    });
                });
        } else {
            resposta.status(405).json({
                "status": false,
                "mensagem": "Método não permitido!"
            });
        }
    }
}

File: Controller/InfoCtrl.py
from model.info import Info

class InfoCtrl:
    # Traduzir comandos HTTP em ações negociais
    # Conceito REST
    # Considerar o protocolo HTTP

    def gravar(self, requisicao):
        if requisicao.method == "POST" and requisicao.is_json:
            dados = requisicao.get_json()

            # pseudo validação
            if all(key in dados for key in ["nome", "descricao", "status", "urlImagem"]):
                servico = Info(
                    0,
                    dados["nome"],
                    dados["descricao"],
                    dados["status"],
                    dados["urlImagem"]
                )

                try:
                    servico.gravar()
                    return {
                        "status": True,
                        "mensagem": "Info gravada com sucesso!",
                        "id": servico.id
                    }, 201
                except Exception as error:
                    return {
                        "status": False,
                        "mensagem": f"Erro ao registrar a info: {error}"
                    }, 500
            else:
                return {
                    "status": False,
                    "mensagem": "Informe todos os dados necessários conforme documentação!"
                }, 400
        else:
            return {
                "status": False,
                "mensagem": "Formato não permitido!"
            }, 405

    def alterar(self, requisicao):
        if requisicao.method in ["PUT", "PATCH"] and requisicao.is_json:
            dados = requisicao.get_json()

            # pseudo validação
            if all(key in dados for key in ["id", "nome", "descricao", "status", "urlImagem"]):
                servico = Info(
                    dados["id"],
                    dados["nome"],
                    dados["descricao"],
                    dados["status"],
                    dados["urlImagem"]
                )

                try:
                    servico.alterar()
                    return {
                        "status": True,
                        "mensagem": "Info alterada com sucesso!"
                    }, 200
                except Exception as error:
                    return {
                        "status": False,
                        "mensagem": f"Erro ao alterar a info: {error}"
                    }, 500
            else:
                return {
                    "status": False,
                    "mensagem": "Informe todos os dados necessários conforme documentação!"
                }, 400
        else:
            return {
                "status": False,
                "mensagem": "Formato não permitido!"
            }, 405

    def excluir(self, requisicao):
        if requisicao.method == "DELETE" and requisicao.is_json:
            id = requisicao.view_args.get("id")  # O id deve ser informado na URL

            # pseudo validação
            if id and int(id) > 0:
                info = Info(int(id))

                try:
                    info.excluir()
                    return {
                        "status": True,
                        "mensagem": "Info excluída com sucesso!"
                    }, 200
                except Exception as error:
                    return {
                        "status": False,
                        "mensagem": f"Erro ao excluir a info: {error}"
                    }, 500
            else:
                return {
                    "status": False,
                    "mensagem": "Informe um id válido na URL!"
                }, 400
        else:
            return {
                "status": False,
                "mensagem": "Formato não permitido!"
            }, 405

    def consultar(self, requisicao):
        termo_busca = requisicao.view_args.get("info")  # Recupera o termo da URL
        if requisicao.method == "GET":
            info = Info(0)

            try:
                lista_infos = info.consultar(termo_busca)
                return {
                    "status": True,
                    "listaInfos": lista_infos
                }, 200
            except Exception as error:
                return {
                    "status": False,
                    "mensagem": f"Não foi possível recuperar as infos: {error}"
                }, 500
        else:
            return {
                "status": False,
                "mensagem": "Método não permitido!"
            }, 405
